### PR TITLE
RHEL7 Build Fixes/Enhancements

### PR DESCRIPTION
**Pull request by mej**
**_WARNING_:  Work In Progress!  Not yet intended for merge!**

This reflects my work underway to get Charliecloud buildable, functional, and "correct" on RHEL7 systems.  I'm going to try to keep it in a merge-worthy state, but it's still not quite to the point where I'm satisfied enough with it to ask that it be considered for merging.  I mainly wanted to publicize what I'm working on and solicit feedback (and allow others to test/play with it, if desired).

In other words, while it's technically a PR, I'm not actually R'ing that it be P'd yet!  ;-)

Please note that I will periodically be rebasing my branch (`mej/charliecloud@fix/rhel7-build`) against `hpc/charliecloud@master`, so anyone who elects to pull/fork from my tree should be ready to [Recover from an Upstream Rebase](https://git-scm.com/docs/git-rebase#_recovering_from_upstream_rebase)!  :-)

### CODE REVIEW

**Comment by reidpr** [comment]
@mej, thanks, this is helpful. What I'm seeing here is two separate improvements:

* Make RPM builds work better.
* Make it easier to check out an embedded Bats.

I'm interested in both of these. Could you do separate PRs for each topic when you believe the work is ready for merging?

File: Makefile [hunk 1]
```diff
@@ -4,7 +4,7 @@ SHELL=/bin/bash
 export CFLAGS += -std=c11 -Wall
 
 .PHONY: all
-all: VERSION.full bin/version.h bin/version.sh
+all: VERSION.full bin/version.h bin/version.sh charliecloud.spec
```
Comment (reidpr) [inline Makefile]: Right now the package builds are not invoked by `make` at the topic level. If we change this, I'd like to make the change mindfully. What's your reasoning?

File: Makefile [hunk 2]
```diff
@@ -43,15 +43,23 @@ bin/version.h: VERSION.full
 bin/version.sh: VERSION.full
 	echo "version () { echo 1>&2 '$$(cat $<)'; }" > $@
 
+# Clone BATS if it's not already there
```
Comment (reidpr) [inline Makefile]: Bats is a Git submodule, so these should be `git submodule` commands rather than `clone`.

I am pleased to have it encapsulated somehow though, since I never remember the proper syntax.

File: Makefile [hunk 3]
```diff
@@ -61,7 +69,7 @@ export: VERSION.full man/charliecloud.1
 	tar Af main.tar bats.tar
 	tar --xform=s,^,charliecloud-$$(cat VERSION.full)/, \
             -rf main.tar \
-            man/*.1 VERSION.full
+            $^
```
Comment (reidpr) [inline Makefile]: `$^` doesn't capture all the man pages since only `charliecloud.1` is depended upon.

**Comment by reidpr** [comment]
@mej, can you clarify the status on this PR? Are you still intending to request merge at some point? @j-ogas is working on RPMs now.

**Comment by mej** [comment]
I need to revisit; I believe it may no longer be necessary.  Is there a "stalled" state or something?  Otherwise, feel free to close this one; I can always re-open or do a new one if need be.

**Comment by reidpr** [comment]
There's not a stalled state in the current workflow. We do have a "deferred" tag but currently that's used for bugs/issues we don't plan to implement but haven't completely rejected. The subject matter of this PR we do plan to address, though, and I'm skeptical of leaving PRs around without a plan to merge.

I'll close. Please feel free to re-open if/when appropriate.
